src/ESPAsyncWebServer.h

```diff
@@ -271,13 +273,20 @@ class AsyncWebServerRequest {
     bool hasArg(const char* name) const;         // check if argument exists
     bool hasArg(const __FlashStringHelper * data) const;         // check if F(argument) exists
 
-    const String& pathArg(size_t i) const;
-
     const String& header(const char* name) const;// get request header value by name
     const String& header(const __FlashStringHelper * data) const;// get request header value by F(name)    
     const String& header(size_t i) const;        // get request header value by number
     const String& headerName(size_t i) const;    // get request header name by number
     String urlDecode(const String& text) const;
+
+#ifdef _ASYNCWEBSERVER_REGEX
+    private:
+      LinkedList<String *> _pathParams;
+      void _addPathParam(const char *param);
+
+    public:
+      const String& pathArg(size_t i) const;
```

Comment:
this functions should exists either way. just return whatever is the equivalent of error when regex is disabled.

Reply:
in other words... examples that use it, should compile regardless if regex is enabled or not.

Reply:
Personally I'm not a fan of run time errors. Now you get a error on compile time when ASYNCWEBSERVER_REGEX was not defined.

But then again that is my opinion.

Reply:
I've changed it. If ASYNCWEBSERVER_REGEX is not defined it handles the function if the pathArg was not found.

I added a compiler warning, if the function is used but ASYNCWEBSERVER_REGEX was not defined